src/pages/signin/SignInPage.js

```diff
@@ -85,7 +86,11 @@ class SignInPage extends Component {
         // - AND an account did not exist or is not validated for that login
         const showResendValidationLinkForm = this.props.credentials.login && !validAccount;
 
-        const welcomeText = this.props.translate(`welcomeText.${showPasswordForm ? 'phrase4' : 'phrase1'}`);
+        const welcomeText = showResendValidationLinkForm && this.props.account.validationCodeFailedMessage
+            ? ''
+            : this.props.translate(`welcomeText.${showPasswordForm ? 'phrase4' : 'phrase1'}`);
+
+        const resendLinkTitleMessage = this.props.account.validationCodeFailedMessage ?? 'resendValidationForm.weSentYouMagicSignInLink';
```

Comment:
AFAIK we are not widely using the null coalescence operator in our JS code and standardize on `lodashGet()` for many things. This should be something like 

```js
lodashGet(this.props.account, 'validationCodeFailedMessage', 'resendValidationForm.weSentYouMagicSignInLink')
```

But as mentioned elsewhere ... I don't think it's correct to store the translation key.